Advanced-Recruit-Training/src/LogFileReader.cpp

```diff
@@ -57,15 +57,30 @@ bool LogFileReader::readAll(const QString& fileName)
  * that the conversion from string to double is sucessful.*/
 bool LogFileReader::parseLine(const QString& line, BatteryData& batteryData) const
 {
+    bool ok;
     QStringList sections = line.split(BATDATA_DELIMITER);
+    if(sections.size()!=3){
+        return false;
+    }
 
     QString timeString = sections.at(0);
     batteryData.time = QTime::fromString(timeString, STRING_TIME_FORMAT);
 
-    batteryData.voltage = sections.at(1).toDouble();
+    batteryData.voltage = sections.at(1).toDouble(&ok);
 
-    batteryData.current = sections.at(2).toDouble();
+    batteryData.current = sections.at(2).toDouble(&ok);
 
-    return true;
 
+    if(!batteryData.time.isValid()){
+       return false;
+    }
+    if(batteryData.time.isNull()){
```

Comment:
According to the QT docs:
Returns true if the time is null (i.e., the QTime object was constructed using the default constructor); otherwise returns false. _A null time is also an invalid time._

Since a null time is also invalid, the earlier if statement makes this redundant.